logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/webSocketStatus/PersistPersistentWebSocketConnectionStatusUseCase.kt

```diff
@@ -1,33 +1,25 @@
 package com.wire.kalium.logic.feature.user.webSocketStatus
 
-import com.wire.kalium.logger.KaliumLogger.Companion.ApplicationFlow.LOCAL_STORAGE
-import com.wire.kalium.logic.StorageFailure
-import com.wire.kalium.logic.configuration.GlobalConfigRepository
+import com.wire.kalium.logger.KaliumLogger
+import com.wire.kalium.logic.data.session.SessionRepository
 import com.wire.kalium.logic.functional.fold
 import com.wire.kalium.logic.kaliumLogger
 
 interface PersistPersistentWebSocketConnectionStatusUseCase {
-    operator fun invoke(enabled: Boolean): Unit
+    suspend operator fun invoke(enabled: Boolean)
 }
 
 internal class PersistPersistentWebSocketConnectionStatusUseCaseImpl(
-    private val globalConfigRepository: GlobalConfigRepository
+    private val sessionRepository: SessionRepository
 ) : PersistPersistentWebSocketConnectionStatusUseCase {
-    override operator fun invoke(enabled: Boolean): Unit =
-        globalConfigRepository.persistPersistentWebSocketConnectionStatus(enabled).let {
-            it.fold({ storageFailure ->
-                when (storageFailure) {
-                    is StorageFailure.DataNotFound ->
-                        kaliumLogger.withFeatureId(LOCAL_STORAGE).e(
-                            "DataNotFound when persisting web socket status  : $storageFailure"
-                        )
-
-                    is StorageFailure.Generic ->
-                        kaliumLogger.withFeatureId(LOCAL_STORAGE).e(
-                            "Failure when persisting web socket status ", storageFailure.rootCause
-                        )
-                }
-            }, {}
+    override suspend operator fun invoke(enabled: Boolean) {
+        sessionRepository.currentSession().fold({
```

Comment:
Using current session here is incorrect logic since it will always update the setting for the current user, this may work just fine in android since this will be changed from the UI and the UI is always the current user, but it will break for CLI.
You can inject the user id into constructor and move this usecase to the user session scope